IntroOOP/RationalNumbers/RatNum.cs

```diff
@@ -0,0 +1,369 @@
+        // Дополнительный множитель ко второй дроби
+        int additionalMultiplierSecond = leastCommonMultiple / b.Denominator;
+        // Умножение числителей дробей А, В на доп множ
+        x = new RatNum(a.Numerator * additionalMultiplierFirst, a.Denominator * additionalMultiplierFirst);
+        y = new RatNum(b.Numerator * additionalMultiplierSecond, b.Denominator * additionalMultiplierSecond);
+    }
+    // Возвращает сокращенную дробь
+    public RatNum Reduce()
+    {
+        RatNum result = this;
+        int greatestCommonDivisor = GetCommonDivisor(Numerator, Denominator);
+        result.Numerator /= greatestCommonDivisor;
+        result.Denominator /= greatestCommonDivisor;
+        return result;
+    }
+    // Операция сложения дробей
+    private static RatNum PerformOperationAddition(RatNum a, RatNum b)
+    {
+        // Наименьшее общее кратное знаменателей
+        int leastCommonMultiple = GetCommonMultiple(a.Denominator, b.Denominator);
+        // Дополнительный множитель к первой дроби
+        int additionalMultiplierFirst = leastCommonMultiple / a.Denominator;
+        // Дополнительный множитель ко второй дроби
+        int additionalMultiplierSecond = leastCommonMultiple / b.Denominator;
+        // Умножение числителей дробей А, В на доп множ
+        int tempNumeratorA = a.Numerator * additionalMultiplierFirst;
+        int tempNumeratorB = b.Numerator * additionalMultiplierSecond;
+        // Вычисление нового знаменателя для итоговой дроби
+        int CommonDenominator = a.Denominator * additionalMultiplierFirst;
+
+        return new RatNum(tempNumeratorA + tempNumeratorB, CommonDenominator);
+    }
+
+    // Операция вычитания дробей
+    private static RatNum PerformOperationSubtraction(RatNum a, RatNum b)
+    {
+        // Наименьшее общее кратное знаменателей
+        int leastCommonMultiple = GetCommonMultiple(a.Denominator, b.Denominator);
+        // Дополнительный множитель к первой дроби
+        int additionalMultiplierFirst = leastCommonMultiple / a.Denominator;
+        // Дополнительный множитель ко второй дроби
+        int additionalMultiplierSecond = leastCommonMultiple / b.Denominator;
+        // Умножение числителей дробей А, В на доп множ
+        int tempNumeratorA = a.Numerator * additionalMultiplierFirst;
+        int tempNumeratorB = b.Numerator * additionalMultiplierSecond;
+        // Вычисление нового знаменателя для итоговой дроби
+        int CommonDenominator = a.Denominator * additionalMultiplierFirst;
+
+        return new RatNum(tempNumeratorA - tempNumeratorB, CommonDenominator);
+    }
+    // Обратная дробь
+    private RatNum ReverseNum()
+    {
+        return new RatNum(Denominator, Numerator);
+    }
+    // Метод сравнения.
+    public bool Equals(RatNum x)
+    {
+        RatNum a = Reduce();
+        RatNum b = x.Reduce();
+        return a.Numerator == b.Numerator && a.Denominator == b.Denominator;
+    }
```

Comment:
Если дробь - это класс (а в `x`, соответственно ссылка на объект), то в `x` может попасть `null` и обрушить вашу программу на 133 строке. 
```suggestion
    public bool Equals(RatNum x)
    {
        if(x is null) return false;
        RatNum a = Reduce();
        RatNum b = x.Reduce();
        return a.Numerator == b.Numerator && a.Denominator == b.Denominator;
    }
```